Update Dead-Letter Queue Code samples

YvonneCeelie opened the pull request:
Closes https://github.com/AxonFramework/AxonFramework/issues/2394

Commit:
Updated not working code samples

gklijs reviewed (approved):
Looks good to me 👍

smcvb commented on axon-framework/events/event-processors/README.md:
```diff
@@ -307,12 +307,12 @@ A `JpaSequencedDeadLetterQueue` configuration example:
 public class DeadLetterQueueExampleConfig {
     
-    public ConfigurerModule configureDeadLetterQueueFor(String processingGroup) {
+    public ConfigurerModule configureDeadLetterQueueFor(EventProcessingConfigurer configurer) {
```
I think the non-Spring configuration wouldn't have to construct a `ConfigurerModule` at all.
That's a specific solution for within Spring environments to add configuration methods on the fly.

In the Java Config world, I'd expect a `void` method that receives the `EventProcessingConfigurer` upon which it invokes the configuration changes.

smcvb commented on axon-framework/events/event-processors/README.md:
```diff
@@ -331,12 +331,12 @@ public class DeadLetterQueueExampleConfig {
     
     @Autowired 
-    public ConfigurerModule configureDeadLetterQueueFor(String processingGroup) {
+    public ConfigurerModule configureDeadLetterQueueFor(EventProcessingConfigurer configurer) {
```
I don't think we need the `EventProcessingConfigurer` at all, right?
This method returns a `ConfigurerModule`, which is a `Supplier<Configurer>`. 
So, the `configurer` that's used comes natural from the lambda notation.

smcvb commented on axon-framework/events/event-processors/README.md:
```diff
@@ -331,12 +331,12 @@ public class DeadLetterQueueExampleConfig {
 public class DeadLetterQueueExampleConfig {
     
     @Autowired 
```
This should be `@Bean` instead of `@Autowired`.
AF will look for any `ConfigurerModule` beans and tie them into the configuration process.

The `@Autowired` approach (with `void` return type) _used_ to work, but caused cyclic dependencies after some recent update of Spring.

```suggestion
    @Bean 
```

smcvb commented on axon-framework/events/event-processors/README.md:
```diff
@@ -519,9 +519,9 @@ public class EnqueuePolicyConfigurer {
 
     @Bean
-    public ConfigurerModule configureEnqueuePolicy(String processingGroup) {
+    public ConfigurerModule configureEnqueuePolicy(EventProcessingConfigurer configurer) {
```
The `EventProcessingConfigurer` parameter shouldn't be necessary here, due to the nature of the `ConfigurerModule`.

smcvb commented on axon-framework/events/event-processors/README.md:
```diff
@@ -307,12 +307,12 @@ A `JpaSequencedDeadLetterQueue` configuration example:
         return configurer -> configurer.eventProcessing(
             eventProcessingConfigurer -> eventProcessingConfigurer.registerDeadLetterQueue(
-                processingGroup,
+                "MyProcessingGroup",
```
I'm wondering how many people will copy-paste `MyProcessingGroup` into their code and complain it doesn't work...

Perhaps we can add a notice here stating users should supply their actual processing group names?
Maybe a comment would do inside the code block, or a single sentence before the code-block starts.

smcvb commented on axon-framework/events/event-processors/README.md:
```diff
@@ -307,12 +307,12 @@ A `JpaSequencedDeadLetterQueue` configuration example:
+                "MyProcessingGroup",
                 configuration -> JpaSequencedDeadLetterQueue.builder()
-                                                            .processingGroup(processingGroup)
+                                                            .processingGroup("MyProcessingGroup")
```
Nit: the other code sample where we configure something for a Processing Group (the `ListenerInvocationErrorHandler`) uses `"my-processing-group"` instead of `MyProcessingGroup`. 
It would be good if these samples stay consistent with one another.

smcvb reviewed (changes requested):
Couple of remarks here.

smcvb committed:
Add AF 4.6.1 release notes

Add AF 4.6.1 release notes

smcvb committed:
Add Multi-Tenancy extesions 4.6.1 release notes

Add Multi-Tenancy extesions 4.6.1 release notes

smcvb committed:
Add Tracing extesions 4.6.1 release notes

Add Tracing extesions 4.6.1 release notes

smcvb committed:
Add JGroups and MultiTenancy minor release link to summary

Add JGroups and MultiTenancy minor release link to summary

MGathier committed:
add AxonServer SE 4.6.5

MGathier committed:
Merge remote-tracking branch 'origin/4.6' into 4.6

MGathier committed:
add AxonServer EE 4.6.5

OLibutzki committed:
Method name fix: usingPooledStreamingProcessors --> usingPooledStreamingEventProcessors

(cherry picked from commit 19e4ba77016a8a9b573b0139d6b46a7cf5234943)

OLibutzki committed:
Renamed deprecated method: workerExecutorService ->
workerExecutor

(cherry picked from commit 84913ee2e5a0f89b4c57fc5c0f96cf5d870783b5)

smcvb committed:
Append RevisionSnapshotFilter description to FilteringSnapshotEvents

Append RevisionSnapshotFilter description to the Filtering
Snapshot Events section. This should further clarify that all a user has
 to do to enable revision-based snapshot filtering, is add the @Revision
 annotation.

#enhancement/revision-snapshot-filter

smcvb committed:
Add a notice stating why Sagas do not support the DLQ

Add a notice stating why Sagas do not support the DLQ.

#enhancement/saga-dlq-notice

smcvb committed:
Introduce a dedicated token stealing section

Introduce a dedicated token stealing section onto to the Streaming
Processor page. The intent is to guide users further in what token
stealing is, why it happens, when it happens, and what the consequence
may be to them.

#288

smcvb committed:
Add missing link to UnitOfWork

Add missing link to UnitOfWork

#288

smcvb committed:
Replace command sample for third-party service

Replace command sample for third-party service, as command dispatching
would only occur again in a distributed scenario. Although defining this
 wouldn't be off, it's out of the scope of the intent of the enumartion.

#288

smcvb committed:
Update axon-framework/tuning/event-snapshots.md

Co-authored-by: Sara Torrey <sara.torrey@axoniq.io>

smcvb committed:
Merge pull request #293 from AxonIQ/enhancement/revision-snapshot-filter

Append `RevisionSnapshotFilter` description to Filtering Snapshot Events sub-section

smcvb committed:
Update axon-framework/events/event-processors/streaming.md

Co-authored-by: Sara Torrey <sara.torrey@axoniq.io>

smcvb committed:
Update axon-framework/events/event-processors/streaming.md

Co-authored-by: Sara Torrey <sara.torrey@axoniq.io>

smcvb committed:
Update axon-framework/events/event-processors/streaming.md

Co-authored-by: Sara Torrey <sara.torrey@axoniq.io>

smcvb committed:
Update axon-framework/events/event-processors/streaming.md

Co-authored-by: Sara Torrey <sara.torrey@axoniq.io>

smcvb committed:
Update axon-framework/events/event-processors/streaming.md

Co-authored-by: Sara Torrey <sara.torrey@axoniq.io>

smcvb committed:
Update axon-framework/events/event-processors/streaming.md

Co-authored-by: Sara Torrey <sara.torrey@axoniq.io>

smcvb committed:
Replace ascertain for guarantee

Replace ascertain for guarantee for simplicity in language use.

#288

smcvb committed:
Add 4.6.2 release notes

Add 4.6.2 release notes

smcvb committed:
Merge pull request #294 from AxonIQ/enhancement/saga-dlq-notice

Add notices on Dead-Letter Queue support for Sagas

smcvb committed:
Merge pull request #295 from AxonIQ/enhancement/288

[#288] Add Token Stealing subsection to Tracking Tokens

smcvb committed:
Merge remote-tracking branch 'origin/4.6' into dead-letter-example-code

smcvb committed:
Fine tune code samples for DLQ configuration

Fine tune code samples for DLQ configuration

#287

smcvb reviewed (approved):
My concerns have been addressed, hence I'm approving this pull request.
